arch/arm/boot/dts/overlays/revpi-connect4-overlay.dts

```diff
@@ -0,0 +1,262 @@
+
+			pibridge {
+				pinctrl-names = "default";
+				compatible = "kunbus,pibridge";
+				pinctrl-0 = <&digital_io_pins>;
+
+				/* digital IN, digital OUT */
+				connect-gpios = <&gpio 6 GPIO_ACTIVE_HIGH>,
+						<&gpio 16 GPIO_ACTIVE_HIGH>;
+
+				left-sniff-gpios = <&expander_core 8 GPIO_ACTIVE_HIGH>,
+						   <&expander_core 9 GPIO_ACTIVE_HIGH>;
+				right-sniff-gpios = <&expander_core 11 GPIO_ACTIVE_HIGH>,
+						   <&expander_core 12 GPIO_ACTIVE_HIGH>;
+			};
+
+			leds {
+				compatible = "gpio-leds";
+				power_red {
+					label = "power:1:fault";
+					gpios = <&expander_core 0 GPIO_ACTIVE_HIGH>;
+					linux,default-trigger = "power_red";
+				};
+				a1_red {
+					label = "a1:red:status";
+					gpios = <&expander_core 2 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a1_red";
+				};
+				a1_green {
+					label = "a1:green:status";
+					gpios = <&expander_core 1 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a1_green";
+				};
+				a1_blue {
+					label = "a1:blue:status";
+					gpios = <&expander_core 3 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a1_blue";
+				};
+				a2_red {
+					label = "a2:red:status";
+					gpios = <&expander_core 5 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a2_red";
+				};
+				a2_green {
+					label = "a2:green:status";
+					gpios = <&expander_core 4 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a2_green";
+				};
+				a2_blue {
+					label = "a2:blue:status";
+					gpios = <&expander_core 6 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a2_blue";
+				};
+				a3_red {
+					label = "a3:red:status";
+					gpios = <&expander_power 1 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_red";
+				};
+				a3_green {
+					label = "a3:green:status";
+					gpios = <&expander_power 0 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_green";
+				};
+				a3_blue {
+					label = "a3:blue:status";
+					gpios = <&expander_power 2 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_blue";
+				};
+			};
+		};
+	};
+
+	/* PINMUX */
+	fragment@1 {
+		target = <&gpio>;
+		__overlay__ {
+			pb_uart_pins: pb_uart_pins {
+				/* TX RX TX_EN */
+				brcm,pins     = <4 5 7>;
+				brcm,function = <BCM2835_FSEL_ALT4>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			digital_io_pins: digital_io_pins {
+				/* digital IN, digital OUT */
+				brcm,pins     = <6 16>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN
+				                 BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF
+				                 BCM2835_PUD_OFF>;
+			};
+			exp_core_pins: exp_core_pins {
+				brcm,pins     = <8>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			exp_power_pins: exp_power_pins {
+				brcm,pins     = <9>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			i2c1_pins: i2c1_pins {
+				/* SDA, SCL */
+				brcm,pins     = <2 3>;
+				brcm,function = <BCM2835_FSEL_ALT0>;
```

Comment:
The configuration of the pulls are missing

Reply:
brcm,pull = <BCM2835_PUD_UP>; 
if the node is still needed.

Reply:
As we have external pull-up: BCM2835_PUD_OFF

Reply:
Oh, get it, find it in page 18, thx.

for general, as I understand, 
if pull "high" required, and an external pull-up connected, the brcm,pull should be configured as OFF, otherwise UP
if pull "low" required, and an external pull-down connected, the brcm,pull should be configured as OFF, otherwise DOWN
Is that?

Reply:
Yes as we would other wise have two pull-up/downs